streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -114,7 +133,7 @@ public static JoinWindows of(final Duration timeDifference) throws IllegalArgume
     public JoinWindows before(final Duration timeDifference) throws IllegalArgumentException {
         final String msgPrefix = prepareMillisCheckFailMsgPrefix(timeDifference, "timeDifference");
         final long timeDifferenceMs = validateMillisecondDuration(timeDifference, msgPrefix);
-        return new JoinWindows(timeDifferenceMs, afterMs, DEFAULT_GRACE_PERIOD_MS);
+        return new JoinWindows(timeDifferenceMs, afterMs, graceMs, enableSpuriousResultFix);
```

Comment:
As per the feedback from @ableegoldman the grace period in the old methods need to stay at 24h. It is in the new methods that we get to specify it as zero.

Reply:
Sure. Don't see the connection to this change thought? Note, that `before()` and `after()` are non-static method, and thus, they should not change/set the grace period. Only the static `of(size)` and non-static `grace()` should change grace period.
```
JoinWindow.of(5000).before(30); // of() will set default of 24h, so no need for before() to reset to 24h, it can just inherit it
JoinWindows.of(5000).grace(40).before(30); // this should leave grace at `40` however, without this fix `before()` would re-set grace to the default of 24h what is incorrect
```